src/pages/index.js

```diff
@@ -112,21 +89,62 @@ addCardFormValidator.enableValidation();
 const editCardFormValidator = new FormValidator(config, profileEditForm);
 editCardFormValidator.enableValidation();
 
-//Section.js
-
-const cardSection = new Section(
-  {
-    items: initialCards,
-    renderer: (cardData) => {
-      const cardElement = createCard(cardData);
-      cardSection.addItem(cardElement);
-    },
-  },
-  ".cards__list"
+const editAvatarValidator = new FormValidator(config, editAvatarElement);
+editAvatarValidator.enableValidation();
+
+/* -------------------------------------------------------------------------- */
+/*                      Section class and Promise                             */
+/* -------------------------------------------------------------------------- */
+
+let cardSection;
+Promise.all([api.getUserInfo(), api.getInitialCards()]).then(
+  ([data, initialCards]) => {
+    userData.setUserInfo(data.name, data.about);
+    userData.setUserAvatar(data.avatar);
+    cardSection = new Section(
+      {
+        items: initialCards.reverse(),
+        renderer: (cardData) => {
+          const cardElement = createCard(cardData);
+          cardSection.addItem(cardElement);
+        },
+      },
+      ".cards__list"
+    );
+    cardSection.renderItems();
+  }
 );
-cardSection.renderItems();
 
-//popupWithForm.js (Edit Form)
+/* -------------------------------------------------------------------------- */
+/*                           Update Avatar Image                              */
+/* -------------------------------------------------------------------------- */
+
+const editAvatarForm = new PopupWithForm("#profile-image-modal", (avatar) => {
+  editAvatarForm.setActionText(true);
+  // document.querySelector('#modal-avatar-submit').textContent = "Saving...";
+  api
+    .updateAvatar(avatar)
+    .then((userInfo) => {
+      userData.setUserAvatar(userInfo.avatar);
+      // editAvatarForm.setActionText(false);
+      document.querySelector('#modal-avatar-submit').textContent = "Save";
+      editAvatarForm.close();
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => profileForm.setActionText(false));
+});
+editAvatarForm.setEventListeners();
+
+const avatarButton = document.querySelector(".avatar-edit-button");
+avatarButton.addEventListener("click", () => {
+  editAvatarForm.open();
```

Comment:
Please, call `toggleButtonState` when you open the avatar popup to disable the button